core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/ThreeIndexProperties.java

```diff
@@ -28,6 +30,25 @@ public <Type_> Type_ getProperty(int index) {
         }
     }
 
+    @Override
+    public <Type_> Type_ toKey(int length, int startingPosition) {
```

Comment:
(applies to all toKey impls)

All JDK and java ecosystem libraries typically use from (inclusive) and to (exclusive).
The use of (from,  length) is rare. The use of (length, from) I haven't seen before.
Let's follow common JDK and java ecosystem habits and use (from, to) instead of (length, from).

Reply:
Motivation:
List.subList(from, to)
String.substring(beginIndex, endIndex)

Both are inclusive, exclusive. I 'd use "from" and "to" in naming, as it's clearer ("end" is ambiguous when it's exclusive), and because I see it more often.